src/hooks.ts

```diff
@@ -1,85 +1,78 @@
- */
-export async function buildHook (moduleContainer, moduleOptions, logger) {
+import type { ModuleConfiguration, SentryHandlerProxy } from '../types'
+import { clientSentryEnabled, serverSentryEnabled, envToBool, canInitialize } from './utils'
+import { resolveRelease, ResolvedClientOptions, resolveClientOptions, ResolvedServerOptions, resolveServerOptions } from './options'
+
+const RESOLVED_RELEASE_FILENAME = 'sentry.release.config.mjs'
+
+export async function buildHook (nuxt: Nuxt, moduleOptions: ModuleConfiguration, logger: Consola): Promise<void> {
   const release = await resolveRelease(moduleOptions)
 
+  const templateDir = fileURLToPath(new URL('./templates', import.meta.url))
+
   const pluginOptionClient = clientSentryEnabled(moduleOptions) ? (moduleOptions.lazy ? 'lazy' : 'client') : 'mocked'
-  const clientOptions = merge({ config: { release } }, await resolveClientOptions(moduleContainer, moduleOptions, logger))
-  moduleContainer.addPlugin({
-    src: resolve(__dirname, '..', `plugin.${pluginOptionClient}.js`),
-    fileName: 'sentry.client.js',
+  const clientOptions: ResolvedClientOptions = defu({ config: { release } }, await resolveClientOptions(nuxt, moduleOptions, logger))
+  addPluginTemplate({
+    src: resolve(templateDir, `plugin.${pluginOptionClient}.js`),
+    filename: 'sentry.client.js',
     mode: 'client',
     options: clientOptions,
   })
 
   const pluginOptionServer = serverSentryEnabled(moduleOptions) ? 'server' : 'mocked'
-  const serverOptions = merge({ config: { release } }, await resolveServerOptions(moduleContainer, moduleOptions, logger))
-  moduleContainer.addPlugin({
-    src: resolve(__dirname, '..', `plugin.${pluginOptionServer}.js`),
-    fileName: 'sentry.server.js',
+  const serverOptions: ResolvedServerOptions = defu({ config: { release } }, await resolveServerOptions(nuxt, moduleOptions, logger))
+  addPluginTemplate({
+    src: resolve(templateDir, `plugin.${pluginOptionServer}.js`),
+    filename: 'sentry.server.js',
     mode: 'server',
     options: serverOptions,
   })
 
   if (serverSentryEnabled(moduleOptions)) {
-    moduleContainer.addTemplate({
-      src: resolve(__dirname, '..', 'templates', 'options.ejs'),
-      fileName: RESOLVED_RELEASE_FILENAME,
+    addTemplate({
+      src: resolve(templateDir, 'options.ejs'),
+      filename: RESOLVED_RELEASE_FILENAME,
       options: { release },
     })
   }
 
   // Tree shake debugging code if not running in dev mode and Sentry debug option is not enabled on the client.
   if (!clientOptions.dev && !clientOptions.config.debug) {
     const webpack = await import('webpack').then(m => m.default || m)
-    moduleContainer.options.build.plugins = moduleContainer.options.build.plugins || []
-    moduleContainer.options.build.plugins.push(
-      new webpack.DefinePlugin({
-        __SENTRY_DEBUG__: 'false',
-      }),
-    )
+    addWebpackPlugin(new webpack.DefinePlugin({
+      __SENTRY_DEBUG__: 'false',
+    }))
+    // TODO: Handle Vite
```

Comment:
The `addWebpackPlugin` helper [uses `webpack:config` hook](https://github.com/nuxt/nuxt/blob/64b835c70701c8f27debe8a534ec99034bda812b/packages/kit/src/build.ts#L43-L56) running it here might be too late.

Reply:
This works fine because it's done from `before:build` hook.